### ISSUE TITLE
The alternate language versions of the page are still inserted in the header doesn't works

### ISSUE THREAD

**Issue by not-only-code** [not-only-code/qtranslate-slug#25]
Source:
http://kuchenundkakao.wordpress.com/2012/08/10/qtranslate-slug-bugfixing/

The alternate language versions of the page are still inserted in the header by qtranslate, not knowing about the new slug versions

Solution:

``` php

remove_action('wp_head','qtrans_header');

add_action('wp_head','qtranslate_slug_header_extended');

function qtranslate_slug_header_extended(){
  global $qtranslate_slug;
  global $q_config;
  if(is_404()) return;

  foreach($q_config['enabled_languages'] as $language) {
    if($language != qtrans_getLanguage())
      echo '<link hreflang="'.$language.'" href="'.$qtranslate_slug->get_current_url($language).'" rel="alternate" />'."\n";
  }
}
```

**Comment by not-only-code** [comment]
Changes made in version 1.1 with **home_url** solves this bug, then now **href** links shows a properly url per each language and this action has no sense.

I close this topic.

**Comment by LC43** [comment]
i think this problem persists.

this code removes the `meta http-equiv` tag and important qtranslate styles ( like the flags background ).

should we include the styles in our qts-defaults.css file?

otherwise, this is ready to fix a couple of issues!

issue on wp.org:
http://wordpress.org/support/topic/hreflang-creates-broken-link-404

and issue #44

**Comment by LC43** [comment]
as for the meta tag, W3C says to not use it any longer, and rely on <html lang= 

http://www.w3.org/International/questions/qa-http-and-lang.en

**Comment by not-only-code** [comment]
We could add a checkbox in the admin section (let me think where is the best place) if user wants to show the `<link />` tag for his old theme.
I think by default in modern themes `lang` attribute is added in the `<html />` tag.

**Comment by not-only-code** [comment]
About styles, i think styling  a flag with css and background image is a bad solution, because every language generates his own image path dynamically.

This would be more easy if the menu 'image' generates an `<img />` tag inside of `<li />`, [here](https://github.com/not-only-code/qtranslate-slug/blob/trunk/includes/class-qtranslate-slug.php#L2247). The rest of generic styles could go in `qts-defaults.css`.

**Comment by LC43** [comment]
I like your idea! I was thinking of appending the bg image style to the
qts-default.css dynamicly. Maybe another option box is needed because it
may break some sites. People with no knowledge of css might panic.

**Comment by LC43** [comment]
providing sources:
http://www.w3.org/TR/html401/struct/dirlang.html#adef-lang
http://www.w3.org/TR/html401/struct/links.html#adef-hreflang

**Comment by LC43** [comment]
fixed in 1.1.8
